samples/extensions/subgroups_operations/subgroups_operations.cpp

```diff
@@ -0,0 +1,503 @@
+	VkPipelineLayoutCreateInfo pipeline_layout_create_info =
+	    vkb::initializers::pipeline_layout_create_info(
+	        &texture_object.descriptor_set_layout,
+	        1u);
+	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &pipeline_layout_create_info, nullptr, &texture_object.pipeline.pipeline_layout));
+}
+
+void SubgroupsOperations::setup_descriptor_set()
+{
+	VkDescriptorSetAllocateInfo alloc_info =
+	    vkb::initializers::descriptor_set_allocate_info(
+	        descriptor_pool,
+	        &texture_object.descriptor_set_layout,
+	        1u);
+	VK_CHECK(vkAllocateDescriptorSets(get_device().get_handle(), &alloc_info, &texture_object.descriptor_set));
+	VkDescriptorBufferInfo            buffer_descriptor = create_descriptor(*texture_uniform_buffer);
+	VkDescriptorImageInfo             image_descriptor  = create_descriptor(texture_object.texture);
+	std::vector<VkWriteDescriptorSet> write_descriptor_sets =
+	    {
+	        vkb::initializers::write_descriptor_set(
+	            texture_object.descriptor_set,
+	            VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,
+	            0u,
+	            &buffer_descriptor),
+	        vkb::initializers::write_descriptor_set(
+	            texture_object.descriptor_set,
+	            VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
+	            1u,
+	            &image_descriptor)};
+	vkUpdateDescriptorSets(get_device().get_handle(), static_cast<uint32_t>(write_descriptor_sets.size()), write_descriptor_sets.data(), 0u, NULL);
+}
+
+void SubgroupsOperations::load_assets()
+{
+	texture_object.texture = load_texture("textures/vulkan_logo_full.ktx", vkb::sg::Image::Color);
+	generate_quad();
+}
+
+void SubgroupsOperations::build_command_buffers()
+{
+	VkCommandBufferBeginInfo command_buffer_begin_info = vkb::initializers::command_buffer_begin_info();
+
+	std::array<VkClearValue, 2> clear_values;
+	clear_values[0].color        = {{0.0f, 0.0f, 0.0f, 0.0f}};
+	clear_values[1].depthStencil = {0.0f, 0u};
+
+	VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+	render_pass_begin_info.renderPass               = render_pass;
+	render_pass_begin_info.renderArea.extent.width  = width;
+	render_pass_begin_info.renderArea.extent.height = height;
+	render_pass_begin_info.clearValueCount          = static_cast<uint32_t>(clear_values.size());
+	render_pass_begin_info.pClearValues             = clear_values.data();
+
+	for (uint32_t i = 0u; i < draw_cmd_buffers.size(); ++i)
+	{
+		render_pass_begin_info.framebuffer = framebuffers[i];
+
+		VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffers[i], &command_buffer_begin_info));
+
+		vkCmdBeginRenderPass(draw_cmd_buffers[i], &render_pass_begin_info, VK_SUBPASS_CONTENTS_INLINE);
+
+		VkViewport viewport = vkb::initializers::viewport(static_cast<float>(width), static_cast<float>(height), 0.0f, 1.0f);
+		vkCmdSetViewport(draw_cmd_buffers[i], 0u, 1u, &viewport);
+
+		VkRect2D scissor = vkb::initializers::rect2D(width, height, 0, 0);
```

Comment:
Perhaps it can be explicitly cast by static_cast<int32_t>(width) and static_cast<int32_t>(height). This is not necessary, just a small suggestion